sdk/maps/maps-search/samples/v1-beta/typescript/src/search.ts

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates Search API usage. Simple queries are performed.
+ */
+
+import { DefaultAzureCredential } from "@azure/identity";
+import { AzureKeyCredential } from "@azure/core-auth";
+import { GeoJsonLineString, GeoJsonPolygon } from "../../maps-common/src";
+import {
+  MapsSearchClient,
+  StructuredAddress,
+  SearchAddressRequest,
+  ReverseSearchAddressRequest,
+  FuzzySearchRequest,
+  GeoJsonPolygonCollection,
+  GeoJsonCircleOrPolygonFeatureCollection,
+} from "@azure/maps-search";
+import * as dotenv from "dotenv";
+dotenv.config();
+/**
+ * Azure Maps supports two ways to authenticate requests:
+ * - Shared Key authentication (subscription-key)
+ * - Azure Active Directory (Azure AD) authentication
+ *
+ * In this sample you can put MAPS_SUBSCRIPTION_KEY into .env file to use the first approach or populate
+ * the three AZURE_CLIENT_ID, AZURE_CLIENT_SECRET & AZURE_TENANT_ID variables for trying out AAD auth.
+ *
+ * More info is available at https://docs.microsoft.com/en-us/azure/azure-maps/azure-maps-authentication.
+ */
+
+async function main() {
+  let client: MapsSearchClient;
+
+  if (process.env.MAPS_SUBSCRIPTION_KEY) {
+    // Use subscription key authentication
+    const credential = new AzureKeyCredential(process.env.MAPS_SUBSCRIPTION_KEY);
+    client = new MapsSearchClient(credential);
+  } else {
+    // Use Azure AD authentication
+    if (process.env.MAPS_CLIENT_ID) {
+      const credential = new DefaultAzureCredential();
+      const mapsClientId = process.env.MAPS_CLIENT_ID;
+      client = new MapsSearchClient(credential, mapsClientId);
+    } else {
+      throw Error("Cannot authenticate the client.");
+    }
+  }
+
+  console.log(" --- Geocode address:");
+  console.log(await client.searchAddress("400 Broad, Seattle"));
```

Comment:
Need to show how to manipulate the result instead of simply print the result. A good example is like [Storage service](https://github.com/andykao1213/azure-sdk-for-js/blob/andrewkao/maps-search/sdk/storage/storage-blob/samples/v12/typescript/src/listContainers.ts)

Reply:
I add detailed example in `sample-dev/searchAdressResult.ts`. Also add a sample for the batch-request API in `sample-dev/batchRequest.ts`. See 5c023e2f10b8003e978dda4473a03e971f8d5b69 and 01a342f92a28b4cbb9def7e34ce03076e0d9591b